src/components/interface/cbufp/cbuf_c.c

```diff
@@ -0,0 +1,189 @@
+/**
```

Comment:
Again, in the future, we'll want to `cbufp` -> `cbuf`.  If you have the time, do it now, unless it will take a lot of time.